src/usr/bin/diffuse

```diff
@@ -1087,7 +1090,7 @@ class Preferences:
                             entry = Gtk.Entry.new()
                         widgets[tpl[1]] = entry
                         entry.set_text(self.string_prefs[tpl[1]])
-                    table.attach(entry, 1, 2, i, i + 1, Gtk.AttachOptions.EXPAND|Gtk.AttachOptions.FILL, Gtk.AttachOptions.FILL)
+                    table.attach(entry, 1, i, 1, 1)
```

Comment:
I am just a bit concerned about the changes with the `attach` functions above. Before it was using `i` and `i + 1`, I don't understand how it is possible not to be using the same logic now. Could you explain why this should work? Thanks!

Reply:
The arguments of `Gtk.Table.attach()` and Gtk.Grid.attach()` differ somewhat. See the docs [here](https://lazka.github.io/pgi-docs/Gtk-3.0/classes/Table.html#Gtk.Table.attach) and [here](https://lazka.github.io/pgi-docs/Gtk-3.0/classes/Grid.html#Gtk.Grid.attach), respectively. Basically, for grids, we specify a width and height of each child element, instead of the indices of the cells where the right and bottom of the child element should be attached. The arguments for attach options are gone, but I observed no changes in the layout of the preference dialog.